### PR TITLE
test/pylib: minio unique temp dir

### PR DESCRIPTION
**Pull request by alecco**
Create a unique minio server temp dir for each test run.

### CODE REVIEW

**Comment by xemul** [comment]
Since @gleb-cloudius kills its test.py without cleanup, will minio roots accumulate over time on his node?

**Comment by gleb-cloudius** [comment]
> Since @gleb-cloudius kills its test.py without cleanup, will minio roots accumulate over time on his node?

Why do we create the tmp dir in testlog and not /tmp?

**Comment by xemul** [comment]
```
commit 1de69228f18e03153fa472404a5e628e5f9272ec
Author: Konstantin Osipov <kostja@scylladb.com>
Date:   Sat Dec 7 20:37:53 2019 +0300

    test.py: add --tmpdir
    
    It will be used for test log files.

diff --git a/test.py b/test.py
index d871498628..5930893021 100755
--- a/test.py
+++ b/test.py
@@ -192,6 +192,13 @@ def parse_cmd_line():
                 tests starting with "mem" in "boost" suite. Default: run all
                 tests in all suites.""",
     )
+    parser.add_argument(
+        "--tmpdir",
+        action="store",
+        default="testlog",
+        help="""Path to temporary test data and log files. The data is
+        further segregated per build mode. Default: ./testlog.""",
+    )
     parser.add_argument('--mode', choices=all_modes, action="append", dest="modes",
                         help="Run only tests for given build mode(s)")
     parser.add_argument('--repeat', action="store", default="1", type=int,
```

**Comment by xemul** [comment]
Part of this merge

```
commit da9dae3dbe74635336ccbc48dccbef25418c46c5
Merge: e8b659ec5d a665fab306
Author: Pekka Enberg <penberg@scylladb.com>
Date:   Wed Jan 15 12:50:51 2020 +0200

    Merge 'test.py: add support for CQL tests' from Kostja
```

File: test/pylib/minio_server.py [hunk 1]
```diff
@@ -113,4 +115,4 @@ async def stop(self):
         finally:
             self.logger.info('Killed minio server')
             self.cmd = None
-            shutil.rmtree(self.rootdir)
+            shutil.rmtree(self.tempdir)
```
Comment (xemul) [inline test/pylib/minio_server.py:118]: Good catch, btw. There's self.mcdir that's nowadays not cleaned even on stop :\

**Comment by gleb-cloudius** [comment]
What is going on with this? I still suffer.

**Comment by xemul** [comment]
> Since @gleb-cloudius kills its test.py without cleanup, will minio roots accumulate over time on his node?

Because you didn't tell if you're ok with accumulating minio tempdirs (or maybe they won't accumulate)

**Comment by gleb-cloudius** [comment]
> > Since @gleb-cloudius kills its test.py without cleanup, will minio roots accumulate over time on his node?
> 
> Because you didn't tell if you're ok with accumulating minio tempdirs (or maybe they won't accumulate)

I see. Well I prefer simple previous PR to be honest, but accumulating tempdirs will be less annoying than the current state.

**Review by gleb-cloudius — APPROVED** [review]

**Pull request closed without merging**